[ADD] Enable mass_editing module version 10.0

zaoral opened the pull request:
[task#7111](https://www.vauxoo.com/web?#id=7111&view_type=form&model=project.task&menu_id=136&action=138)

Enable mass_editing module to be installable in version 10.0. This also means all the fixed required to be able to run with new api and according to the new model

zaoral committed:
[ADD] Enable mass_editing module to be installable

zaoral committed:
[FIX] API 10.0 update deprecated select field attribute to index

Before this change a ""WARNING openerp_test odoo.fields: Field
mass.object.name: parameter 'select' is no longer supported; use 'ind
ex' instead."" was raising in my odoo instance.

nhomar commented:
@zaoral Once this is merged, please sent the same PR to OCA/server-tools

zaoral committed:
[pylint] Fix C8106 (manifest-version-format)

Wrong Version Format '9.0.1.0.0' in manifest file
New version is 10.0.1.0.0

zaoral committed:
[pylint] Fix E8103 (sql-injection)

Use of "%" operator in execute database method. Better use parameters instead.

nhomar commented on mass_editing/models/mass_object.py:
```diff
@@ -10,7 +10,7 @@ class MassObject(models.Model):
     _description = "Mass Editing Object"
 
-    name = fields.Char('Name', required=True, select=1)
+    name = fields.Char('Name', required=True, index=1)
```
remove 'Name' parameter, not necesary on v10 after the fact it is taking field name on Camel Case if no string attribute (just for this feald no refactor all please).

zaoral replied:
Applied, currently resolving another errors in the unit tests.

moylop260 replied:
FYI we have developed a pylint check for this case.
I just enabled from 
https://github.com/Vauxoo/pylint-conf/commit/3a63d5917ddf6b85f82022fb80605dd84ed9a734

cc @JesusZapata

zaoral replied:
+1

zaoral committed:
[pylint] Fix W8102 (copy-wo-api-one)

Missing api.one or api.multi in copy function: added api.multi

zaoral committed:
[pylint] Fix W7950 (odoo-addons-relative-import)

Same Odoo module absolute import. You should use relative import with
"." instead of "openerp.addons.mass_editing"

zaoral committed:
[REF] API 10.0 remove 'Name' string attribute in the field definition with the
same name

Not necessary on v10 after the fact it is taking field name on Camel Case
if no string attribute.

zaoral committed:
[FIX] Warning ir.actions.act_window.create() includes unknown fields:
auto_refresh

No longer supported field 'auto_refresh' as mentioned in commit
8be378150171c4a5a60b82e13421b2915d9b3024

zaoral committed:
[FIX] WARNING openerp_test odoo.models: module_prototyper.create() includes unknown fields: dependencies

Update the reference of this field in the unit tests form dependencies
to dependency_ids.

moylop260 commented on mass_editing/hooks.py:
```diff
@@ -8,5 +8,5 @@ def uninstall_hook(cr, registry):
     for res in cr.dictfetchall():
         value = 'ir.actions.act_window,%s' % res.get('id')
-        cr.execute("DELETE FROM ir_values WHERE value = '%s'" % value)
+        cr.execute("DELETE FROM ir_values WHERE value = '%s'", (value,))
```
`s/'%s'/%s/g`

zaoral replied:
Thanks @moylop260

zaoral committed:
[FIX] ERROR test_uninstall_hook

ProgrammingError: syntax error at or near "ir"
 LINE 1: DELETE FROM ir_values WHERE value = 'ir.actions.act_window,...

This is fixed updating the sentence in the hook.

zaoral commented:
@moylop260 @nhomar 
I think This PR is ready to be merged

> **NOTE:** We have Warning in the travis tests but are not related to this module that are shown as green results.
>
> - module base_external_dbsource
   ![image](https://cloud.githubusercontent.com/assets/7593953/22703762/e67cb618-ed2a-11e6-8e38-c80c944f4607.png)
> - module  auto_backup
   ![image](https://cloud.githubusercontent.com/assets/7593953/22703994/bf44e574-ed2b-11e6-8460-821fdb23f8cb.png)
> - module password_security
   ![image](https://cloud.githubusercontent.com/assets/7593953/22703883/4d3f214c-ed2b-11e6-9867-87145842e20b.png)

zaoral commented:
FYI, we have a green build in the dummy PR https://git.vauxoo.com/vauxoo/instance/merge_requests/358
